lib/dynflow/dispatcher.rb

```diff
@@ -15,14 +15,20 @@ module Dispatcher
         fields! receiver_id: String
       end
 
-      variants Event, Execution, Ping
+      Steps = type do
+        fields! receiver_id: String,
+                execution_plan_id: type { variants String, NilClass }
+      end
+
+      variants Event, Execution, Ping, Steps
     end
 
     Response = Algebrick.type do
       variants Accepted = atom,
                Failed   = type { fields! error: String },
                Done     = atom,
-               Pong     = atom
+               Pong     = atom,
+               PendingSteps = type { fields! pending_steps: Hash }
```

Comment:
I'm not sure if `Steps` is the best word, as it has already some meaning: the items there might be either steps or events for the steps: it would be more like `execution items`

Reply:
True, I'll try to come up with better name for it